test/OpLoopControlINTELGDB.spt

```diff
@@ -0,0 +1,100 @@
+119734787 65536 393230 58 0
+2 Capability Addresses
+2 Capability Kernel
+2 Capability GenericPointer
+2 Capability Int8
+2 Capability UnstructuredLoopControlsINTEL
+11 Extension "SPV_INTEL_unstructured_loop_controls"
+5 ExtInstImport 1 "OpenCL.std"
+5 ExtInstImport 2 "SPIRV.debug"
+3 MemoryModel 2 2
+7 EntryPoint 6 5 "_ZTSZ4mainE3foo"
+11 String 27 "kernel_arg_type._ZTSZ4mainE3foo."
+23 String 28 "intel-fpga-loops.cpp"
+3 String 33 ""
+5 String 36 "operator()"
+6 String 43 "_ZTSZ4mainE3foo"
+7 String 47 "_ZZ4mainENK3$_0clEv"
+4 String 49 "this"
+3 Source 4 100000
+9 Name 7 "class._ZTSZ4mainE3$_0.anon"
+7 Name 18 "_ZZ4mainENK3$_0clEv"
+4 Decorate 10 Alignment 1
+4 Decorate 25 Alignment 8
+4 TypeInt 8 8 0
+4 TypeInt 34 32 0
+4 Constant 34 35 8
+2 TypeVoid 3
+3 TypeFunction 4 3
+3 TypeStruct 7 8
+4 TypePointer 9 7 7
+4 TypePointer 11 7 8
+4 TypePointer 15 8 7
+4 TypeFunction 17 3 15
+4 TypePointer 24 7 15
+
+5 ExtInst 3 13 2 DebugInfoNone
+7 ExtInst 3 29 2 DebugSource 28 13
+9 ExtInst 3 30 2 DebugCompileUnit 65536 4 29 4
+7 ExtInst 3 31 2 DebugTypeFunction 0 13
+7 ExtInst 3 37 2 DebugTypeQualifier 32 0
+8 ExtInst 3 38 2 DebugTypePointer 37 4294967295 288
+8 ExtInst 3 39 2 DebugTypeFunction 0 13 38
+13 ExtInst 3 40 2 DebugFunctionDecl 36 39 29 17 0 32 33 8323
+15 ExtInst 3 32 2 DebugTypeComposite 33 0 29 17 0 30 13 35 32768 40
+8 ExtInst 3 42 2 DebugTypePointer 37 4294967295 0
+16 ExtInst 3 44 2 DebugFunction 43 31 29 17 0 30 33 8360 0 5 13
+12 ExtInst 3 45 2 DebugLocalVariable 33 32 29 0 0 44 0
+9 ExtInst 3 46 2 DebugLexicalBlock 29 0 0 44
+16 ExtInst 3 48 2 DebugFunction 36 39 29 17 0 32 47 8332 17 18 40
+7 ExtInst 3 50 2 DebugSource 33 13
+13 ExtInst 3 51 2 DebugLocalVariable 49 42 50 0 0 48 288 1
+5 ExtInst 3 52 2 DebugExpression
+
+5 Function 3 5 0 4
+
+2 Label 6
+4 Variable 9 10 7
+4 Bitcast 11 12 10
+3 LifetimeStart 12 1
+6 ExtInst 3 53 2 DebugScope 44
+4 Line 28 0 0
+8 ExtInst 3 14 2 DebugDeclare 45 10 52
+6 ExtInst 3 54 2 DebugScope 46
+4 PtrCastToGeneric 15 16 10
+5 FunctionCall 3 20 18 16
+6 ExtInst 3 55 2 DebugScope 44
+4 Bitcast 11 21 10
+3 LifetimeStop 21 1
+1 Return
+
+1 FunctionEnd
+
+5 Function 3 18 0 17
+3 FunctionParameter 15 19
+
+2 Label 22
+4 Variable 24 25 7
+5 Store 25 19 2 8
+6 ExtInst 3 56 2 DebugScope 48
+4 Line 28 0 0
+8 ExtInst 3 26 2 DebugDeclare 51 25 52
+4 Line 28 19 5
+2 Branch 23
+
+2 Label 23
+6 ExtInst 3 57 2 DebugScope 48
+2 LoopControlINTEL 1
+4 Line 28 19 5
+2 Branch 23
+
+1 FunctionEnd
+
+; RUN: llvm-spirv %s -to-binary -o %t.spv
```

Comment:
Why is there no check for LLVM --> SPIR-V translation?

Reply:
Because the patch fixes a bug in LLVM IR consumption. I'm sure, that LLVM --> SPIR-V translation with debug information is good covered in DebugInfo/DebugControlFlow.cl test.

Reply:
Reworked, thanks!